### PR TITLE
executor: fix leak of tiflashTestSuite

### PR DESCRIPTION
**Pull request by tiancaiamao**
### What problem does this PR solve?

Problem Summary:

### What is changed and how it works?

What's Changed:

Fix a leak in the test code.

How it Works:

Binary search to locate the case
https://github.com/pingcap/tidb/pull/26712#issuecomment-892354357

### Check List 

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [x] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note 

```release-note
None
```

### CODE REVIEW

File: executor/tiflash_test.go [hunk 1]
```diff
@@ -75,6 +75,11 @@ func (s *tiflashTestSuite) SetUpSuite(c *C) {
 	s.dom.SetStatsUpdating(true)
 }
 
+func (s *tiflashTestSuite) TearDownSuite(c *C) {
+	s.dom.Close()
+	s.store.Close()
```
Comment (xhebox) [inline executor/tiflash_test.go:80]: Maybe assert the return value?

**Review by wjhuang2016 — APPROVED** [review]

**Review by tisonkun — APPROVED** [review]
Second @xhebox suggestion, we will turn on `errcheck` sooner or later.

Rest LGTM.

File: executor/tiflash_test.go (hunk 1 above)
Comment (tiancaiamao) [reply executor/tiflash_test.go]: Done

**Review by xhebox — APPROVED** [review]

**Pull request merged**